src/components/approval-type/ApprovalType.vue

```diff
@@ -0,0 +1,157 @@
+          <!-- COURT ORDER radio button -->
+          <v-radio id="court-order-radio" class="mb-0"
+             :label="getRadioText(ApprovalTypes.VIA_COURT_ORDER)"
+             :value="ApprovalTypes.VIA_COURT_ORDER"
+          />
+          <v-form ref="courtNumRef" id="court-num-form" v-model="valid" class="mt-4 ml-8">
+            <v-text-field
+              @input="courtOrderNumberChanged"
+              id="court-order-number-input"
+              v-model="courtOrderNumber"
+              label="Court Order Number"
+              :rules="courtOrderNumRules"
+              :disabled="approvalTypeSelected === ApprovalTypes.VIA_REGISTRAR"
+              @update:error="emitValidationError($event)"
+              filled
+            />
+          </v-form>
+          <!-- REGISTRAR radio button -->
+          <v-radio id="registrar-radio" class="mb-0 pt-2"
+             :label="getRadioText(ApprovalTypes.VIA_REGISTRAR)"
+             :value="ApprovalTypes.VIA_REGISTRAR"
+             v-if="!isCourtOrderOnly"
+          />
+        </v-radio-group>
+      </v-col>
+    </v-row>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import { Component, Emit, Prop } from 'vue-property-decorator'
+import { FormIF } from '@bcrs-shared-components/interfaces'
+
+@Component({})
+export default class ApprovalType extends Vue {
+  // Refs
+  $refs!: Vue['$refs'] & {
+    courtNumRef: FormIF
+  }
+
+  ApprovalTypes = {
+    VIA_COURT_ORDER: 'VIA COURT ORDER',
+    VIA_REGISTRAR: 'VIA REGISTRAR'
+  }
+
+  /** Draft court order number. */
+  @Prop({ default: '' }) readonly draftCourtOrderNumber!: string
+
+  /** Whether approved by the registrar. */
+  @Prop({ default: false }) readonly draftApprovedByRegistrar!: boolean
+
+  /** filing name used in radio options. */
+  @Prop({ default: 'restoration'}) readonly filingType!: string
+
+  /** Show only the court order option; remove via registrar option. */
+  @Prop({ default: false }) readonly isCourtOrderOnly!: boolean
+
+  // Local properties
+  private courtOrderNumber = ''
+  private courtOrderNumRules = []
+  private valid = false
+  private approvalTypeSelected = ''
+
+  /** Called when component is mounted. */
+  mounted (): void {
+    // Copy props to mutable properties
+    if (this.draftApprovedByRegistrar) {
+      this.courtOrderNumber = ''
+      this.approvalTypeSelected = this.ApprovalTypes.VIA_REGISTRAR
+    }
+    if (this.draftCourtOrderNumber) {
+      this.courtOrderNumber = this.draftCourtOrderNumber
+      this.approvalTypeSelected = this.ApprovalTypes.VIA_COURT_ORDER
+    }
+  }
+
+  /** Triggers the form validation. */
+  public validate (): boolean {
+    let status = this.$refs.courtNumRef.validate()
+    this.$emit('emitValid', status)
+    return status
+  }
+
+  private emitValidationError (event): void {
+    this.$emit('emitValid', !event)
+  }
+
+  @Emit('emitRadioButtonChange')
+  private radioButtonChanged (event): void {
```

Comment:
types on the arguments might be helpful here.

Reply:
If it were me, I'd name the first one "valid" and the second one "value" (or val).

Reply:
I agree, that's better.